odoo/addons/base/models/ir_model.py

```diff
@@ -623,6 +623,26 @@ def _onchange_related(self):
             self.relation = field.relation
             self.readonly = True
 
+    def _validate_relation_field(self):
+        model_name = self.relation
+        model_id = self.env['ir.model']._get_id(model_name)
+        if not model_id:
+            raise UserError(_("Unknown model name '%s' in relation model") % (model_name))
+
+    @api.onchange('relation')
+    def _onchange_relation(self):
+        if self.relation:
+            try:
+                self._validate_relation_field()
+            except UserError as e:
+                return {'warning': {'title': _("Warning"), 'message': e}}
+
+    @api.constrains('relation')
+    def _check_relation(self):
+        for rec in self:
+            if rec.state == 'manual' and rec.relation:
+                rec._validate_relation_field()
```

Comment:
This method should raise a `ValidationError`.  Also please inline `_validate_relation_field` inside this method, and simply use `_check_relation` inside the onchange method.  Having a third method has no added value here.